htdocs/admin/limits.php

```diff
@@ -66,29 +72,53 @@
 
     if (! $error)
     {
-        dolibarr_set_const($db, "MAIN_MAX_DECIMALS_UNIT", $_POST["MAIN_MAX_DECIMALS_UNIT"],'chaine',0,'',$conf->entity);
-        dolibarr_set_const($db, "MAIN_MAX_DECIMALS_TOT", $_POST["MAIN_MAX_DECIMALS_TOT"],'chaine',0,'',$conf->entity);
-        dolibarr_set_const($db, "MAIN_MAX_DECIMALS_SHOWN", $_POST["MAIN_MAX_DECIMALS_SHOWN"],'chaine',0,'',$conf->entity);
+    	dolibarr_set_const($db, $mainmaxdecimalsunit, $_POST[$mainmaxdecimalsunit],'chaine',0,'',$conf->entity);
+    	dolibarr_set_const($db, $mainmaxdecimalstot, $_POST[$mainmaxdecimalstot],'chaine',0,'',$conf->entity);
+    	dolibarr_set_const($db, $mainmaxdecimalsshown, $_POST[$mainmaxdecimalsshown],'chaine',0,'',$conf->entity);
 
-        dolibarr_set_const($db, "MAIN_ROUNDING_RULE_TOT", $_POST["MAIN_ROUNDING_RULE_TOT"],'chaine',0,'',$conf->entity);
+    	dolibarr_set_const($db, $mainroundingruletot, $_POST[$mainroundingruletot],'chaine',0,'',$conf->entity);
 
-        header("Location: ".$_SERVER["PHP_SELF"]."?mainmenu=home&leftmenu=setup");
+        header("Location: ".$_SERVER["PHP_SELF"]."?mainmenu=home&leftmenu=setup".(! empty($currencycode)?'&currencycode='.$currencycode:''));
         exit;
     }
 }
 
 
-
 /*
  * View
-*/
+ */
 
 $form=new Form($db);
 
 llxHeader();
 
 print load_fiche_titre($langs->trans("LimitsSetup"),'','title_setup');
 
+if (! empty($conf->multicurrency->enabled) && ! empty($conf->global->MULTICURRENCY_USE_LIMIT_BY_CURRENCY))
+{
+	require_once DOL_DOCUMENT_ROOT.'/core/lib/multicurrency.lib.php';
+
+	$currencycode = (! empty($currencycode)?$currencycode:$conf->currency);
+	$aCurrencies = array($conf->currency);	// Default currency always first position
+
+	$sql = 'SELECT rowid, code FROM '.MAIN_DB_PREFIX.'multicurrency';
+	$sql.= ' WHERE entity = '.$conf->entity;
+	$sql.= ' AND code != "'.$conf->currency.'"';	// Default currency always first position
+	$resql = $db->query($sql);
+	if ($resql)
+	{
+		while ($obj = $db->fetch_object($resql))
+		{
+			$aCurrencies[] = $obj->code;
+		}
+
+		if (! empty($aCurrencies) && count($aCurrencies) > 1)
+		{
+			$head = multicurrencyLimitPrepareHead($aCurrencies);
+			dol_fiche_head($head, $currencycode, '', -1, "multicurrency");
```

Comment:
You introduce here a dol_fiche_head, you should probably introduce a dol_fiche_end too on same condition, shouldn't it ?

Reply:
@eldy ok, but did you receive my question about these changes?

Reply:
@eldy 
the trouble is that the functions that integrate these constants, do not
do not get back, the code of the currency!
do you see a solution without modifying all the "prices",
"price2num", "calcul_price_total" and other codes using these constants?